components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceDataHolder.java

```diff
@@ -529,4 +531,14 @@ public void setServerSessionManagementService(
 
         this.serverSessionManagementService = sessionManagementService;
     }
+
+    public void setAccountLockService(AccountLockService accountLockService) {
+
+        this.accountLockService = accountLockService;
+    }
+
+    public AccountLockService getAccountLockService() {
+
+        return accountLockService;
```

Comment:
Lets validate whether the `accountLockService` is not or not. If null, lets throw an runtime exception

Reply:
In here https://github.com/wso2/carbon-identity-framework/pull/3598/files#diff-8bccdb72ac4633653cb8821508b1c64219f85e428f8bf61fe51c7540c8946b4fR793 this service is marked as mandatory. Service won't be activated if AccountLockService is not set. Why do we need to throw a runtime exception?

Reply:
Agree, I have seen in some places where we have done that as a practice feel to ack that